designs/Transitive-Dependencies.md

```diff
@@ -0,0 +1,150 @@
+- Drilling down the dependency tree
+- In the package detail view, some way to explore dependencies of dependencies and then their dependencies and so on, any level deep, without installing the package.
+
+
+## Cross-cutting Concerns
+
+-	The words “top-level” & “transitive” are words most people must stop & think about.
+    -	Should we use “direct” & “indirect”? Or “explicit” & “implicit”?
+-	Should a transitive dependency have the “installed” text given it’s already technically installed? Should it be called “promote” instead?
+- Given that the tooltip is not accessible for .NET Framework workloads, we will have to come up with an alternative experience or cut the tooltip experience until we have an accessible experience.
+
+## Q&A
+
+- Q: Can I see transitive dependencies on solution view or only project view?
+- A: The scope today is project view first & seeing how complex including this in solution view would be in the engineering costing.
+
+- Q: Who should ultimately own this work?
+- A: The NuGet client team should own this work. They are familiar with the Visual Studio Package Manager UI & concepts of transitive dependencies.
+
+- Q: Should packages.config be considered in this work?
+- A: Since packages.config is based on top-level packages only, it will just use its existing experience by placing all packages in the top-level packages section.
+
+- Q: I want to know what transitive dependencies are being brought in by a package. How do I do that?
+- A: You install package X & view the transitive dependencies panel inside Visual Studio or within the project system’s NuGet reference node. If we do CLI work, you would be able to type “dotnet list package –include-transitive” to see a similar output.
+
+- Q: I want to know all the packages that depend on a transitive package. How do I do that?
+- A: You would use the tooltip to hover over a transitive dependency & view the list of packages that bring it in. Otherwise you would use the dotnet CLI and view the “referenced” column.
+
+- Q: Any thought towards seeing a visual graph for debugging transitive dependencies?
+- A: Tools exist for this such as DGML, https://github.com/ThomasArdal/NuGetPackageVisualizer, & https://docs.microsoft.com/en-us/nuget/release-notes/nuget-1.4#package-visualizer. It is not in the scope of this feature & would likely need to be a part of “debugging dependencies” efforts.
+
+- Q: Any indicator of transitive dependencies in search? We show top-level packages that are installed.
+- A: No because these packages are not technically installed nor managed by the user. The user can only manage top-level dependencies. We want users to be able to promote to a top-level, but we do not need to let them know that something is already transitively brought in.
+
+- Q: Is this really a feasible solution for license auditing? If 99 transitive dependencies are brought in, am I supposed to click through them all?
```

Comment:
When installing, licenses from *all* packages being installed, including transitive dependencies are displayed in the license dialog.

Reply:
Thanks for the clarification! Nothing changes in this regard then.

Reply:
For "Promote" we are awaiting another UX workshop to go over these small details such as what language to be used. We want to remain consistent, but promote probably has a better connotation than "installing" it. Thanks to @skofman1 for the suggestion!

Reply:
Top-level and transitive are things I don't believe we can change although the UX team would like to review them as they are "stop-and-think" words.